src/server/detail/snapshot_storage.cc

```diff
@@ -256,16 +256,19 @@ io::Result<std::string, GenericError> AwsS3SnapshotStorage::LoadPath(std::string
     // Sort the keys in reverse so the first. Since the timestamp format
     // has lexicographic order, the matching snapshot file will be the latest
     // snapshot.
-    io::Result<std::vector<std::string>, GenericError> keys = ListObjects(bucket_name, prefix);
+    io::Result<std::vector<SnapStat>, GenericError> keys = ListObjects(bucket_name, prefix);
     if (!keys) {
       return nonstd::make_unexpected(keys.error());
     }
 
-    std::sort(std::rbegin(*keys), std::rend(*keys));
-    for (const std::string& key : *keys) {
+    std::sort(std::rbegin(*keys), std::rend(*keys), [](const SnapStat& l, const SnapStat& r) {
+      return l.last_modified < r.last_modified;
+    });
+
+    for (const SnapStat& key : *keys) {
       std::smatch m;
-      if (std::regex_match(key, m, re)) {
-        return std::string(kS3Prefix) + bucket_name + "/" + key;
+      if (std::regex_match(key.name, m, re)) {
+        return std::string(kS3Prefix) + bucket_name + "/" + key.name;
```

Comment:
It's better to use `StrCat()` for such cases, it saves many allocations (here it'll be 1 instead of 4 IIUC)

Reply:
(I realize it was here before you touched it, but still why not improve while here :)

Reply:
I will fix it next time when do some changes for this file)